src/crypto/CHIPCryptoPALPSA.cpp

```diff
@@ -1684,6 +1685,128 @@ CHIP_ERROR ExtractAKIDFromX509Cert(const ByteSpan & certificate, MutableByteSpan
+            VerifyOrExit(result == 0, error = CHIP_ERROR_NOT_FOUND);
+            VerifyOrExit(p + len == end_of_ext, error = CHIP_ERROR_NOT_FOUND);
+
+            // The DistributionPointName is:
+            //     DistributionPointName ::= CHOICE {
+            //         fullName                [0]     GeneralNames,
+            //         nameRelativeToCRLIssuer [1]     RelativeDistinguishedName }
+            //
+            // The URI should be encoded in the fullName element.
+            result = mbedtls_asn1_get_tag(&p, end, &len, MBEDTLS_ASN1_CONTEXT_SPECIFIC | MBEDTLS_ASN1_CONSTRUCTED | 0);
+            VerifyOrExit(result == 0, error = CHIP_ERROR_NOT_FOUND);
+
+            //     GeneralNames ::= SEQUENCE SIZE (1..MAX) OF GeneralName
+            result = mbedtls_asn1_get_tag(&p, end, &len, MBEDTLS_ASN1_CONTEXT_SPECIFIC | MBEDTLS_ASN1_CONSTRUCTED);
+            VerifyOrExit(result == 0, error = CHIP_ERROR_NOT_FOUND);
+
+            // The CDP URI is encoded as a uniformResourceIdentifier field of the GeneralName:
+            //     GeneralName ::= CHOICE {
+            //         otherName                       [0]     OtherName,
+            //         rfc822Name                      [1]     IA5String,
+            //         dNSName                         [2]     IA5String,
+            //         x400Address                     [3]     ORAddress,
+            //         directoryName                   [4]     Name,
+            //         ediPartyName                    [5]     EDIPartyName,
+            //         uniformResourceIdentifier       [6]     IA5String,
+            //         iPAddress                       [7]     OCTET STRING,
+            //         registeredID                    [8]     OBJECT IDENTIFIER }
+            result =
+                mbedtls_asn1_get_tag(&p, end, &len, MBEDTLS_ASN1_CONTEXT_SPECIFIC | MBEDTLS_X509_SAN_UNIFORM_RESOURCE_IDENTIFIER);
+            VerifyOrExit(result == 0, error = CHIP_ERROR_NOT_FOUND);
+
+            const char * urlptr = reinterpret_cast<const char *>(p);
+            VerifyOrExit((len > strlen(kValidCDPURIHttpPrefix) &&
+                          strncmp(urlptr, kValidCDPURIHttpPrefix, strlen(kValidCDPURIHttpPrefix)) == 0) ||
+                             (len > strlen(kValidCDPURIHttpsPrefix) &&
+                              strncmp(urlptr, kValidCDPURIHttpsPrefix, strlen(kValidCDPURIHttpsPrefix)) == 0),
+                         error = CHIP_ERROR_NOT_FOUND);
+            error = CopyCharSpanToMutableCharSpan(CharSpan(urlptr, len), cdpurl);
+            SuccessOrExit(error);
+        }
+        p = end_of_ext;
+    }
+
+    VerifyOrExit(cdpExtCount == 1, error = CHIP_ERROR_NOT_FOUND);
+
+exit:
+    logMbedTLSError(result);
+    mbedtls_x509_crt_free(&mbed_cert);
+
+#else
+    (void) certificate;
+    (void) cdpurl;
+    CHIP_ERROR error = CHIP_ERROR_NOT_IMPLEMENTED;
+#endif // defined(MBEDTLS_X509_CRT_PARSE_C)
+
+    return error;
```

Comment:
It looks like this function won't compile if MBEDTLS_X509_CRT_PARSE_C is not defined, since error would be undefined.